### PR TITLE
Use the HTML sanitizer component as Twig filter

### PR DESCRIPTION
**Pull request by ausi**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -121,6 +121,7 @@
         "symfony/filesystem": "^5.4 || ^6.0",
         "symfony/finder": "^5.4 || ^6.0",
         "symfony/framework-bundle": "^5.4 || ^6.0",
+        "symfony/html-sanitizer": "^5.4 || ^6.0",
```
Comment (m-vo) [inline composer.json]:
```suggestion
        "symfony/html-sanitizer": "^6.1",
```

The component was first released in version `6.1`.

Can we maybe also bump the "symfony/twig-bridge" to ^6.1? Then we would also automatically get the [`HtmlSanitizerExtension`](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Bridge/Twig/Extension/HtmlSanitizerExtension.php) that does the setup for the `sanitize_html` filter for free.

Comment (ausi) [reply composer.json]: Done in 55f82e25a04a4de064ad3c328a4ce9cd9861f921.

File: core-bundle/src/Twig/Runtime/SanitizerRuntime.php [hunk 2]
```diff
@@ -12,66 +12,25 @@
...
-    public function __construct(private readonly ContaoFramework $framework)
+    public function __construct(private readonly HtmlSanitizerExtension $sanitizerExtension)
     {
     }
 
-    public function sanitizeHtml(string $html): string
+    public function sanitizeHtml(string $html, string $sanitizer = null): string
     {
-        $this->framework->initialize();
+        $html = $this->sanitizerExtension->sanitize($html, $sanitizer);
```
Comment (m-vo) [inline core-bundle/src/Twig/Runtime/SanitizerRuntime.php]: This looks good to me, just a thought:

In the `ContaoExtension`, for referencing the `EscaperExtension` we inject `@twig` and use `$escaperExtension = $environment->getExtension(EscaperExtension::class);`. Should we do the same here or also inject the explicit service at the other place?

A point for using `@twig` is, that it's independent of service name changes or refactoring of the Twig/bridge setup (happened before). But this version maybe looks a bit cleaner.

**Comment by m-vo** [comment]
@ausi I raised the "symfony/service-contracts" requirements as well in 3505db0, this seems to have made the CI green again. The issue originates from here: https://github.com/symfony/symfony/pull/45680.

IMHO this is a Symfony bug, so maybe we should add a conflict via `contao/conflicts` instead and wait for it to be fixed.

**Comment by m-vo** [comment]
> IMHO this is a Symfony bug, so maybe we should add a conflict via contao/conflicts instead and wait for it to be fixed.

Ok, after some more investigation: Yes it is a bug, no Symfony won't fix this anymore. It's a dependency problem baked in forever. So we need to resolve it ourselves anyway.

File: core-bundle/src/Twig/Runtime/SanitizerRuntime.php (hunk 2 above)
Comment (ausi) [reply core-bundle/src/Twig/Runtime/SanitizerRuntime.php]: Done in 55242b72cab09a5a7a5d42f8e27f3c1d8e0385c0

File: core-bundle/src/Twig/Runtime/SanitizerRuntime.php [hunk 3]
```diff
@@ -12,66 +12,26 @@
...
-        ['key' => 'style', 'value' => 'media'],
-        ['key' => 'time', 'value' => 'datetime'],
-        ['key' => 'details', 'value' => 'open'],
-    ];
-
     /**
      * @internal
      */
-    public function __construct(private readonly ContaoFramework $framework)
+    public function __construct(readonly private Environment $twig)
```
Comment (fritzmg) [inline core-bundle/src/Twig/Runtime/SanitizerRuntime.php]:
```suggestion
    public function __construct(private readonly Environment $twig)
```

**Review by m-vo — DISMISSED** [review]
Big thanks! :heart:

File: composer.json [hunk 4]
```diff
@@ -200,6 +201,8 @@
         "phpunit/phpunit": "<8.0",
         "roave/better-reflection": "<4.12.2 || >=6.0",
         "symfony/dependency-injection": "5.4.16 || 6.0.16 || 6.1.8 || 6.2.0 || 6.2.1",
+        "symfony/framework-bundle": "6.1.0 || 6.1.1",
+        "symfony/twig-bundle": "6.1.0",
```
Comment (leofeyer) [inline composer.json]: These conflicts are unnecessary. We can require `^6.1.2` and `^6.1.1` above instead.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**